tracer/src/Datadog.Trace/AgentProcessManager.cs

```diff
@@ -69,8 +69,9 @@ public static void Initialize()
                 var azureAppServiceSettings = new ImmutableAzureAppServiceSettings(GlobalConfigurationSource.Instance);
 
                 var automaticTraceEnabled = EnvironmentHelpers.GetEnvironmentVariable(ConfigurationKeys.TraceEnabled, string.Empty)?.ToBoolean() ?? true;
+                var automaticProfilingEnabled = ContinuousProfiler.Profiler.Instance.Status.IsProfilerEnabled;
```

Comment:
We tend to read only the env variables because here we're called from the loader and we don't want to load the whole instance. Looking at your code, I think it's ok. But could there be any issues if we change the `ContinuousProfiler.Profiler.Instance` initialization later? Would it be safer to just create a ProfilerStatus instance? 

This reminded me that you need to change another place. Indeed, you should impact the loader [here](https://github.com/DataDog/dd-trace-dotnet/blob/5150a2defb5d3a08161a8fc724dcb8693fc4f55b/tracer/src/Datadog.Trace.ClrProfiler.Managed.Loader/Startup.cs#L55)

Reply:
Yeah, this currently has implications like initializing all the `DatadogLogging` infrastructure etc, which we don't really want to do here. I think it would be safest to change this to a simple env var check

Reply:
I do not see any specific issue, but yeah, maybe reading the env var would be simpler and would avoid any race we might hit later.